rx_signal_processing/rx_dsp_chain.cu

```diff
@@ -15,406 +15,238 @@
+  //can be done in real-time.
+  zmq::socket_t timing_socket(sig_proc_context, ZMQ_PAIR);
+  ERR_CHK_ZMQ(timing_socket.bind("ipc:///tmp/feeds/4"))
+
+
+  auto gpu_properties = get_gpu_properties();
+  print_gpu_properties(gpu_properties);
+
+  uint32_t first_stage_dm_rate = 0, second_stage_dm_rate = 0, third_stage_dm_rate = 0;
+  //Check for non integer dm rates
+  if (fmod(rx_rate,sig_options.get_first_stage_sample_rate()) > 0.0) {
+    //TODO(keith): handle error
+  } //TODO(keith): not sure these checks will work.
+/*  else if (fmod(sig_options.get_first_stage_sample_rate(),
+          sig_options.get_second_stage_sample_rate()) > 0.0) {
+    //TODO(keith): handle error
+  }
+  else if(fmod(sig_options.get_second_stage_sample_rate(),
+        sig_options.get_third_stage_sample_rate()) > 0.0) {
+    //TODO(keith): handle error
+  }*/
+  else{
+    auto float_dm_rate = rx_rate/sig_options.get_first_stage_sample_rate();
+    first_stage_dm_rate = static_cast<uint32_t>(float_dm_rate);
+
+    float_dm_rate = sig_options.get_first_stage_sample_rate()/
+          sig_options.get_second_stage_sample_rate();
+    second_stage_dm_rate = static_cast<uint32_t>(float_dm_rate);
+
+    float_dm_rate = sig_options.get_second_stage_sample_rate()/
+          sig_options.get_third_stage_sample_rate();
+    third_stage_dm_rate = static_cast<uint32_t>(float_dm_rate);
+  }
+
+  std::cout << "1st stage dm rate: " << first_stage_dm_rate << std::endl
+    << "2nd stage dm rate: " << second_stage_dm_rate << std::endl
+    << "3rd stage dm rate: " << third_stage_dm_rate << std::endl;
+
+
+  auto filter_timing_start = std::chrono::steady_clock::now();
+
+  Filtering filters(rx_rate,sig_options);
+
+  DEBUG_MSG("Number of 1st stage taps: " << filters.get_num_first_stage_taps() << std::endl
+    << "Number of 2nd stage taps: " << filters.get_num_second_stage_taps() << std::endl
+    << "Number of 3rd stage taps: " << filters.get_num_third_stage_taps() <<std::endl
+    << "Number of 1st stage taps after padding: "
+    << filters.get_first_stage_lowpass_taps().size() << std::endl
+    << "Number of 2nd stage taps after padding: "
+    << filters.get_second_stage_lowpass_taps().size() << std::endl
+    << "Number of 3rd stage taps after padding: "
+    << filters.get_third_stage_lowpass_taps().size());
+
+  auto filter_timing_end = std::chrono::steady_clock::now();
+  auto time_diff = std::chrono::duration_cast<std::chrono::microseconds>(filter_timing_end -
+                                                                       filter_timing_start).count();
+  DEBUG_MSG("Time to create 3 filters: " << time_diff << "us");
+
+  //FIXME(Keith): fix saving filter to file
+  filters.save_filter_to_file(filters.get_first_stage_lowpass_taps(),"filter1coefficients.dat");
+  filters.save_filter_to_file(filters.get_second_stage_lowpass_taps(),"filter2coefficients.dat");
+  filters.save_filter_to_file(filters.get_third_stage_lowpass_taps(),"filter3coefficients.dat");
```

Comment:
I recall discussing it but not the details. I guess if it works then we're OK? I think we must have discussed something about saving metadata along with the filter, to prevent confusion about when/where it was used.